### ISSUE TITLE
NHWC tensor format  Tensorflow

### ISSUE THREAD

**Issue by sonfiree** [onnx/onnx-tensorflow#535]
`The Conv2D op currently only supports the NHWC tensor format on the CPU. The op was given the format: NCHW`

How to make a convert?

**Comment by TemiLee** [comment]
I also confused in this question, do you have solved ?

**Comment by sonfiree** [comment]
No

**Comment by cong235** [comment]
Because you are running on CPU. Have 3 way for you: 1.Install tensorflow build with mkl, it support ncwh, 2. modify your code to change input datatype from nhwc to ncwh, 3. Run your model on GPU

**Comment by sonfiree** [comment]
1. Not suitable
2. Yes, this is correct, but I did not find a guide on how to do it
3. It does not work

**Comment by cong235** [comment]
1.Easy way:https://pypi.org/project/intel-tensorflow/
2. That is your problem with your code. Try to find where you feed data and what node you pass data with the wrong format.
3.You mean your problem still meets? or you can't run your model on GPU?

**Comment by sonfiree** [comment]
1. This installer does not suit me. I have TF serving model
3. Yes

**Comment by mgarbade** [comment]
I assume the NHWC problem has been fixed by using tensorflow 2.x and the corresponding (most recent) version of onnx and onnx-tf.
However in the most recent version other problems occur, at least when trying to convert from pytorch -> onnx -> tf frozen graph -> tflite (which is probably what most people want).
See #614 and #617

**Comment by sonfire86** [comment]
unfortunately I do not use tf2.X because of bugs.

**Comment by mgarbade** [comment]
True, I was recently examining the tensorflow frozen graph build from onnx and it still complains about the NHWC problem.

**Comment by pushpendrapratap** [comment]
setting `os.environ["CUDA_VISIBLE_DEVICES"]="-1"` (as mentioned in this reply [https://github.com/onnx/onnx-tensorflow/issues/614#issuecomment-625251872](https://github.com/onnx/onnx-tensorflow/issues/614#issuecomment-625251872)) fixes the issue for me.

diff. versions of library I'm using:
```
torch==1.6.0
torchvision==0.7.0
onnx==1.7.0
onnxmltools==1.7.0
tensorflow==2.3.0
tensorflow-addons==0.11.0
onnxruntime==1.4.0
git+https://github.com/onnx/onnx-tensorflow.git (currently, it's onnx-tf==1.6.0)
```

**Comment by shaqing** [comment]
i meet the same poblem, have u fixed it?

**Comment by sonfire86** [comment]
@shahirdaya not fixed

**Comment by shahirdaya** [comment]
@sonfire86 I did not work on this so not sure.  Thanks

**Comment by turned2670** [comment]
> 1.Easy way:https://pypi.org/project/intel-tensorflow/ 2. That is your problem with your code. Try to find where you feed data and what node you pass data with the wrong format. 3.You mean your problem still meets? or you can't run your model on GPU?

The first way fixes my same problem. Thanks.

**Comment by robbinc91** [comment]
> The first way fixes my same problem. Thanks.

[turned2670](https://github.com/turned2670) I can't make it work. Would you please explain how exactly did you solve the problem?
